### PR TITLE
Bump sigstore/sigstore, cosign, in-toto-golang

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

- Bumps s/s to 1.6
- Bumps cosign to v2
- Bumps intoto to latest commit to pull in https://github.com/in-toto/in-toto-golang/commit/2d81ebf6a457bd210a5c4a2b54b0d523c1366894
- Increases timeout for failed pipeline run test. Cosign v2 introduces retries for transparency log writing, which I think is pushing us past the 1 minute threshold we currently have set.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
Chains updated to use cosign v2
```

### CODE REVIEW

**Comment by wlynch** [comment]
e2e test failure because of 1.23 being EOL and not meeting the MIN_K8S version for knative pkg anymore. https://github.com/tektoncd/chains/pull/746 to remove it from the e2e tests.

**Comment by chuangw6** [comment]
Now we can upgrade intoto to v0.7.1. https://github.com/in-toto/in-toto-golang/releases/tag/v0.7.1

**Comment by wlynch** [comment]
> Now we can upgrade intoto to v0.7.1. https://github.com/in-toto/in-toto-golang/releases/tag/v0.7.1

Done.

**Review by chuangw6 — APPROVED** [review]
Thank you @wlynch !

File: pkg/chains/signing/wrap.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func (w *sslSigner) SignMessage(payload io.Reader, opts ...signature.SignOption)
 	if err != nil {
 		return nil, err
 	}
-	env, err := w.wrapper.SignPayload(in_toto.PayloadType, m)
+	env, err := w.wrapper.SignPayload(context.TODO(), in_toto.PayloadType, m)
```
Comment (lcarva) [inline pkg/chains/signing/wrap.go:120]: Can we make `SignMessage` take a context? It looks like the only place that calls this method is from `pkg/chains/signing.go:Sign` which does have a context.

(Ok to do as a follow up)

**Review by lcarva — APPROVED** [review]
/lgtm

**Pull request merged**